modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -293,8 +294,14 @@ public ObjectDefinition deleteObjectDefinition(
 		_objectFieldPersistence.removeByObjectDefinitionId(
 			objectDefinition.getObjectDefinitionId());
 
-		// TODO Deleting an object definition should delete related object
-		// relationships
+		List<ObjectRelationship> objectRelationships =
+			_objectRelationshipLocalService.getObjectRelationships(
+				objectDefinition.getObjectDefinitionId());
+
+		for (ObjectRelationship objectRelationship : objectRelationships) {
+			_objectRelationshipLocalService.deleteObjectRelationship(
+				objectRelationship.getObjectRelationshipId());
+		}
```

Comment:
@danvb112 we should implement a method called `deleteObjectRelationshipByObjectDefinition` in the objectRelationshipLocalService. I think @p-albuquerque did something similar these days...

Reply:
Hi @gabrielwas, I talked to @p-albuquerque and we concluded that, his solution will also solve this bug.

Reply:
@danvb112 Thanks for the updates :D